typescript/sdk/src/token/types.ts

```diff
@@ -147,6 +151,51 @@ export const WarpRouteDeployConfigSchema = z
     });
 
     return z.NEVER; // Causes schema validation to throw with above issue
+  })
+  // Verify that CCIP hooks are paired with CCIP ISMs
+  .transform((warpRouteDeployConfig, ctx) => {
+    // Get all the CCIP hooks and ISMs by origin->destination chain
+    const { ccipHookMap, ccipIsmMap } = getCCIPConfigMaps(
+      warpRouteDeployConfig,
+    );
+
+    // Check that each CCIP hook has a corresponding ISM
+    for (const [originChain, hooks] of Object.entries(ccipHookMap)) {
+      if (!ccipIsmMap[originChain]) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: `No CCIP ISM found in config for origin chain ${originChain}`,
+        });
+        return z.NEVER;
+      }
+
+      for (const [destinationChain] of Object.entries(hooks)) {
```

Comment:
If you are only going to take the keys, imo use `Object.keys`? However since you also are taking the value, why not `[destination, doesExist]`. But then again, when would the `destinationChain` be in here, but the boolean false? Wonder if it would make it simpler to have the inner `ChainMap` as an array or set?

Reply:
changed it from ChainMap<ChainMap<boolean>> into a ChainMap<Set<ChainName>> as suggested, also fixed naming as requested